component.js

```diff
@@ -119,8 +119,8 @@ function factory (options) {
   ComponentCreator.shouldComponentUpdate = _shouldComponentUpdate;
   return ComponentCreator;
 
-  function ComponentCreator (displayName, mixins, render) {
-    var options = createDefaultArguments(displayName, mixins, render);
+  function ComponentCreator (displayName, mixins, render, decorate) {
```

Comment:
I'm sceptical of adding another parameter to the list. I'm not completely convinced, but I think I'd rather have a additional factory function that takes decorators as first argument instead.